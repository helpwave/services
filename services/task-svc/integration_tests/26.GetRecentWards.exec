#!/bin/bash

echo "First, get a clean slate:"
cat <<EOF | redis-cli > /dev/null
DEL recently-viewed-patient-18159713-5d4e-4ad5-94ad-fbb6bb147984
DEL recently-viewed-ward-18159713-5d4e-4ad5-94ad-fbb6bb147984
EOF


grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "51aca99b-2372-40b4-99ed-355dad6c014d"}' \
	localhost:3001 \
	proto.services.task_svc.v1.WardService/UpdateWard > /dev/null

sleep 1

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "626e8c70-dc93-44fb-a9d5-0b828a0a0c4b"}' \
	localhost:3001 \
	proto.services.task_svc.v1.WardService/UpdateWard > /dev/null

sleep 1

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{}' \
	localhost:3001 \
	proto.services.task_svc.v1.WardService/GetRecentWards

