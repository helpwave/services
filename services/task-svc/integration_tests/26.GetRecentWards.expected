First, get a clean slate:
{
  "wards": [
    {
      "id": "51aca99b-2372-40b4-99ed-355dad6c014d",
      "name": "Portage",
      "bedCount": 8,
      "tasksTodo": 7,
      "tasksInProgress": 2,
      "tasksDone": 4
    },
    {
      "id": "626e8c70-dc93-44fb-a9d5-0b828a0a0c4b",
      "name": "Arrowood"
    }
  ]
}
